stingray/events.py

```diff
@@ -709,10 +774,76 @@ def to_astropy_table(self):
     @staticmethod
     def from_astropy_table(ts):
         kwargs = dict([(key.lower(), val) for (key, val) in ts.meta.items()])
-        for attr in ['time', 'energy', 'pi']:
-            if attr in ts.colnames:
-                kwargs[attr] = ts[attr]
+        ev = EventList(time=ts["time"], **kwargs)
+        array_attrs = ts.colnames
+
+        for attr in array_attrs:
+            if attr == "time":
+                continue
+            setattr(ev, attr, ts[attr])
+
+        return ev
+
+    def to_xarray(self):
+        from xarray import Dataset
+        data = {}
+        array_attrs = self.array_attrs()
+
+        for attr in array_attrs:
+            data[attr] = np.asarray(getattr(self, attr))
+
+        ts = Dataset(data)
+
+        ts.attrs['gti'] = self.gti
+        ts.attrs['mjdref'] = self.mjdref
+        ts.attrs['instr'] = self.instr
+        ts.attrs['mission'] = self.mission
+        ts.attrs['header'] = self.header
```

Comment:
Just checking: are these attributes always set?

Reply:
Should be. But I devise a (probably) better method